src/sentry/api/endpoints/project_tagkey_details.py

```diff
@@ -34,6 +35,13 @@ def delete(self, request: Request, project, key) -> Response:
             {method} {path}
 
         """
+        if ratelimiter.is_limited(
+            "api.rate-limit.project-tag-delete",
+            limit=1,
+            window=1,
+        ):
+            return Response({"detail": "Rate limit exceeded."}, status=429)
```

Comment:
This may not matter if you're switching over to the RPS rate limiter, but we should considering switching to using `rest_framework.exceptions` for error responses. In this case `rest_framework.exceptions.Throttled` should convert to a 429 error through DRF